hyperspy/io_plugins/protochips.py

```diff
@@ -0,0 +1,183 @@
+        return {'data': self._data_dictionary[quantity],
+                'axes': self._get_axes(),
+                'metadata': self._get_metadata(quantity),
+                'original_metadata': {'Protochips_header':
+                                      self._get_original_metadata()},
+                'mapping': self._get_mapping()}
+
+    def _get_original_metadata(self):
+        d = {'Start time': self.start_datetime}
+        d['Time units'] = self.time_units
+        for quantity in self.logged_quantity_name_list:
+            d['%s_units' % quantity] = self._get_quantity_units(quantity)
+        d['User'] = self.user
+        d['Calibration file name'] = self._parse_calibration_file()
+        return d
+
+    def _get_mapping(self):
+        return {'Protochips_header.date': ("General.time", None), }
+
+    def _get_metadata(self, quantity):
+        return {'General': {'original_filename': os.path.split(self.filename)[1],
+                            'title': '%s (%s)' % (quantity,
+                                                  self._get_quantity_units(quantity)),
+                            'user': self.user,
+                            'start_time': self.start_datetime,
+                            'notes': self._parse_notes(),
+                            'calibration_file': self._parse_calibration_file()},
+                "Signal": {'signal_type': quantity,
+                           'time_axis': self._get_metadata_time_axis()}}
+
+    def _get_metadata_time_axis(self):
+        return {'value': self.time_axis,
+                'units': self.time_units}
+
+    def _read_data(self, header_line_number):
+        names = [name.replace(' ', '_') for name in self.column_name]
+        data = np.genfromtxt(self.filename, delimiter=',', dtype=None,
+                             names=names,
+                             skip_header=header_line_number,
+                             unpack=True)
+
+        self._data_dictionary = dict()
+        for i, name, name_dtype in zip(range(len(names)), self.column_name,
+                                       names):
+            if name == 'Notes':
+                self.notes = data[name_dtype].astype(str)
+            elif name == 'Time':
+                self.time_axis = data[name_dtype]
+            else:
+                self._data_dictionary[name] = data[name_dtype]
+
+    def _parse_notes(self):
+        arr = np.vstack((self.time_axis, self.notes))
+        return np.compress(arr[1] != '', arr, axis=1)
+
+    def _parse_calibration_file(self):
+         # for the gas cell, the calibration is saved in the notes colunm
+        if self.calibration_file is None:
+            calibration_file = "The calibration files names are saved in "\
+                "metadata.General.notes"
+        else:
+            calibration_file = self.calibration_file
+        return calibration_file
+
+    def _get_axes(self):
+        scale = np.diff(self.time_axis).mean()
```

Comment:
One of the main limitations of HyperSpy is that it doesn't support non-linear axes. We should add support for them asap, but, in the meantime, your approach seems to me like a good compromise. Instead of raising an error if the difference is too large I would add e.g. max(diff) to the log and maybe add a comment about this in the code and the User Guide. Once we add support for non-linear axes we can then deal with this properly.